### ISSUE TITLE
Build from source fails to run "assertion failed: validate_scale_factor(scale_factor) "

### ISSUE THREAD

**Issue by popey** [squidowl/halloy#244]
Hello! Thanks for making & maintaining a beautiful IRC client in the 21st century. :heart: 

The pre-built binary works fine on my system, but a binary built from source doesn't. :confused: 

I cloned the repo and built the release binary, both from tip of git, and a checkout of 2024.1. They both crash on launch with:

```
alan@nuc:~/Source/squidowl/halloy$ target/release/halloy                                                                                                                                                  [46/1123]
thread 'main' panicked at /home/alan/.cargo/git/checkouts/winit-57d3141eaf559308/b91e39e/src/dpi.rs:432:9:                                                                                                         
assertion failed: validate_scale_factor(scale_factor)                                                                                                                                                              
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace         
```

Backtrace:

```
alan@nuc:~/Source/squidowl/halloy$ RUST_BACKTRACE=1 target/release/halloy                                                                                                                                          
thread 'main' panicked at /home/alan/.cargo/git/checkouts/winit-57d3141eaf559308/b91e39e/src/dpi.rs:432:9:                                                                                                         
assertion failed: validate_scale_factor(scale_factor)                                                                                                                                                              
stack backtrace:                            
   0: rust_begin_unwind                                                                                                                                                                                            
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt                                                                                                                                                                                   
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/panicking.rs:72:14
   2: core::panicking::panic                                                                                                                                                                                       
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/panicking.rs:144:5
   3: iced_winit::conversion::window_event                                                                                                                                                                         
   4: core::ops::function::impls::<impl core::ops::function::FnMut<A> for &mut F>::call_mut                                                                                                                        
   5: winit::platform_impl::platform::x11::EventLoop<T>::drain_events::{{closure}}                                                                                                                                 
   6: winit::platform_impl::platform::x11::event_processor::EventProcessor<T>::process_event                                                                                                                       
   7: winit::platform_impl::platform::x11::EventLoop<T>::run_on_demand    
   8: iced_winit::application::run                                                                       
   9: halloy::main                                                                                       
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

The full backtrace:

```
alan@nuc:~/Source/squidowl/halloy$ RUST_BACKTRACE=full target/release/halloy                                                                                                                                       
thread 'main' panicked at /home/alan/.cargo/git/checkouts/winit-57d3141eaf559308/b91e39e/src/dpi.rs:432:9:                                                                                                                                                                                                                   
assertion failed: validate_scale_factor(scale_factor)                                                                                                                                                              
stack backtrace:                                                                                                                                                                                                   
   0:     0x557448431916 - std::backtrace_rs::backtrace::libunwind::trace::hbee8a7973eeb6c93                                                                                                                       
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/../../backtrace/src/backtrace/libunwind.rs:104:5                                                                 
   1:     0x557448431916 - std::backtrace_rs::backtrace::trace_unsynchronized::hc8ac75eea3aa6899                                                                                                                                                                                
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5                                                                                                                                                                                                                                                  
   2:     0x557448431916 - std::sys_common::backtrace::_print_fmt::hc7f3e3b5298b1083                                                                                                                                                                                            
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys_common/backtrace.rs:68:5                                                                                                                                                                                                                                                               
   3:     0x557448431916 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hbb235daedd7c6190                                                                                    
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys_common/backtrace.rs:44:22                                                                                    
   4:     0x557448461a50 - core::fmt::rt::Argument::fmt::h76c38a80d925a410                                                                                                                                                                                                      
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/fmt/rt.rs:142:9                                                                                                 
   5:     0x557448461a50 - core::fmt::write::h3ed6aeaa977c8e45                                                                                                                                                                                                                  
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/fmt/mod.rs:1120:17                                                                                              
   6:     0x55744842d81f - std::io::Write::write_fmt::h78b18af5775fedb5                                                                                                                                                                                                                                                      
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/io/mod.rs:1810:15                                                                                                
   7:     0x5574484316f4 - std::sys_common::backtrace::_print::h5d645a07e0fcfdbb                                                                                                                                                                                                                                             
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys_common/backtrace.rs:47:5                                                                                     
   8:     0x5574484316f4 - std::sys_common::backtrace::print::h85035a511aafe7a8                                                                                                                                    
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys_common/backtrace.rs:34:9 
   9:     0x557448432e47 - std::panicking::default_hook::{{closure}}::hcce8cea212785a25                                                                                                                                                                                                                                      
  10:     0x557448432ba9 - std::panicking::default_hook::hf5fcb0f213fe709a                                                                                                                                         
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:292:9                                                                                                                                                                                                                                                                         
  11:     0x5574484332d8 - std::panicking::rust_panic_with_hook::h095fccf1dc9379ee                                                                                                                                 
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:779:13                                                                                                                                                                                                                                                                        
  12:     0x557448433179 - std::panicking::begin_panic_handler::{{closure}}::h032ba12139b353db                                                                                                                     
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:649:13                                                                                                                                                                                                        
  13:     0x557448431e16 - std::sys_common::backtrace::__rust_end_short_backtrace::h9259bc2ff8fd0f76                                                                                                               
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys_common/backtrace.rs:171:18                                                                                                                                                                                                                                                             
  14:     0x557448432f10 - rust_begin_unwind                                                                                                                                                                       
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:645:5                                                                                                                                                                                                                                                                         
  15:     0x5574477d9cb5 - core::panicking::panic_fmt::h784f20a50eaab275                                                                                                                                                                                                        
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/panicking.rs:72:14                                                                                              
  16:     0x5574477d9d73 - core::panicking::panic::hb837a5ebbbe5b188                                                                                                                                                                                                            
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/panicking.rs:144:5                                                                                                                                                                                                                                                                        
  17:     0x557447b4306a - iced_winit::conversion::window_event::he0c513b894acd23f                                                                                                                                                                                              
  18:     0x557447886ede - core::ops::function::impls::<impl core::ops::function::FnMut<A> for &mut F>::call_mut::h238b005428baef09                                                                                                                                                                                                                                                          
  19:     0x557447919ed9 - winit::platform_impl::platform::x11::EventLoop<T>::drain_events::{{closure}}::he72fe7d71dc47f2a                                                                                                                                                                                                   
  20:     0x557447912319 - winit::platform_impl::platform::x11::event_processor::EventProcessor<T>::process_event::h56452cea75eff383                                                                                                                                                                                                                                                         
  21:     0x557447841d1f - winit::platform_impl::platform::x11::EventLoop<T>::run_on_demand::h21749bf0f3bb072c                                                                                                                                                                                                               
  22:     0x5574478766ce - iced_winit::application::run::h4e26e653f6b3a4fd                                                              
  23:     0x55744793f26f - halloy::main::h7e4bc034b2805716                                                                                                                                                                                                                                                                   
  24:     0x557447976973 - std::sys_common::backtrace::__rust_begin_short_backtrace::h2c3c4db33b6e710b                                                        
  25:     0x557447976992 - std::rt::lang_start::{{closure}}::h11b6074e1378cde3                                                                                                                                                                                                                                               
  26:     0x5574484273b1 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h37600b1e5eea4ecd                                                                                                                                                                                                                                                            
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/ops/function.rs:284:13                                                                                                                                                                                                                                                                    
  27:     0x5574484273b1 - std::panicking::try::do_call::hb4bda49fa13a0c2b                                                                                    
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:552:40                                                                                                                                                                                                                                                                        
  28:     0x5574484273b1 - std::panicking::try::h8bbf75149211aaaa                                                                                             
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:516:19                                                                                                                                                                                                                                                                        
  29:     0x5574484273b1 - std::panic::catch_unwind::h8c78ec68ebea34cb                                                                                                                        
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panic.rs:142:14                                                                                                                                                                                                                                                                            
  30:     0x5574484273b1 - std::rt::lang_start_internal::{{closure}}::hffdf44a19fd9e220                                                                                                       
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/rt.rs:148:48                                                                                                                                                                                                                                                                               
  31:     0x5574484273b1 - std::panicking::try::do_call::hcb3194972c74716d                                                                                                                    
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:552:40                                                                                                                                                                                                                                                                        
  32:     0x5574484273b1 - std::panicking::try::hcdc6892c5f0dba4c                                                                                                                             
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:516:19                                                                                                                                                                                                                                                                        
  33:     0x5574484273b1 - std::panic::catch_unwind::h4910beb4573f4776                                                                                                                        
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panic.rs:142:14                                                                                                                                                                                                                                                                            
  34:     0x5574484273b1 - std::rt::lang_start_internal::h6939038e2873596b                                                                                                                    
                               at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/rt.rs:148:20                                                                                                                                                                                                                                                                               
  35:     0x55744794b665 - main                                                                
  36:     0x7f752c628150 - __libc_start_call_main                                              
                               at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16                                                                                                          
  37:     0x7f752c628209 - __libc_start_main_impl                                              
                               at ./csu/../csu/libc-start.c:360:3                                                                                                                             
  38:     0x5574477da7a5 - _start                                                              
  39:                0x0 - <unknown>  
```

**Comment by casperstorm** [comment]
I see, it might be related to: https://github.com/squidowl/halloy/blob/main/config.yaml#L54
Can you post your config?

**Comment by casperstorm** [comment]
@popey alrite, its fixed now in #245! 
Thanks for reporting this as it only happened without a config file 😮‍💨 
I'll merge it now and then you should be able to run it from source!
